tests/test_lua_details.cc

```diff
@@ -0,0 +1,174 @@
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<bool>(*result) == true);
+    }
+
+    SECTION("return false")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return false");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<bool>(*result) == false);
+    }
+
+    SECTION("return 42")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return 42");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<int>(*result) == 42);
+    }
+
+    SECTION("return 4.2")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return 4.2");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<double>(*result) == 4.2);
+    }
+
+    SECTION("return 'pippo'")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "return 'pippo'");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<std::string>(*result) == "pippo");
+    }
+}
+
+TEST_CASE("execute_lua_script_safely(): Lua exceptions", "[lua_details]")
+{
+    sol::state lua;
+    open_safe_library_subset(lua); // for error() and pcall()
+
+    SECTION("Syntax error")
+    {
+        auto result_or_error = execute_lua_script_safely(lua, "not a lua program");
+        auto* msg = std::get_if<std::string>(&result_or_error);
+        REQUIRE(msg != nullptr);
+        REQUIRE(*msg != "");
+    }
+
+    SECTION("Runtime error")
+    {
+        auto result_or_error = execute_lua_script_safely(
+            lua, "function boom(); error('pippo', 0); end; boom()");
+        auto* msg = std::get_if<std::string>(&result_or_error);
+        REQUIRE(msg != nullptr);
+        REQUIRE_THAT(*msg, StartsWith("Script execution error: pippo"));
+        // Lua adds a stack trace after this output. This is a somewhat brittle test,
+        // but since we have control over our Lua version, we are sure to spot it if
+        // the output format changes.
+    }
+
+    SECTION("Runtime error, caught by pcall()")
+    {
+        auto result_or_error = execute_lua_script_safely(
+            lua, "function boom(); b = nil; b(); end; pcall(boom); return 42");
+        auto* result = std::get_if<sol::protected_function_result>(&result_or_error);
+        REQUIRE(result != nullptr);
+        REQUIRE(static_cast<int>(*result) == 42);
+    }
+}
+
+TEST_CASE("execute_lua_script_safely(): C++ exceptions", "[Step]")
+{
+    sol::state lua;
+    open_safe_library_subset(lua); // for error() and pcall()
+
+    lua["throw_logic_error"] = []() { throw std::logic_error("pippo"); };
```

Comment:
I would always use different words in different tests; makes it far easier to see where some message has been created from. (I mean `"pippo"` here)

Reply:
Fair enough, done.